Fixing bug with qconfig check

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #68908

Summary: The assert_valid_qconfig function tries to check whether a
qconfig on a convT model has a per_channel weight qscheme, but the NS
debug qconfig: torch.ao.quantization.QConfig(activation=logger_cls, weight=None)
results in an error when qconfig.weight() would be called. Added a convt
model to NS test to check for this in the future.

Test Plan: python test/test_quantization.py TestEagerModeNumericSuite.test_compare_model_stub_conv_static

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32928449](https://our.internmc.facebook.com/intern/diff/D32928449)

HDCharles committed:
Getting NS to work with Ferraris

Summary: WIP

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on torch/ao/quantization/qconfig.py:
```diff
@@ -249,7 +249,7 @@ def assert_valid_qconfig(qconfig: Optional[Union[QConfig, QConfigDynamic]],
     Verifies that this `qconfig` is valid.
     """
-    if qconfig is None:
+    if qconfig is None or qconfig.weight is None:
```
might be cleaner to move the weight check inside the if statement, something like

```
if is_conv_transpose_mod:
    if qconfig.weight is None:
      # for now, we assume that any qconfig for ConvTranspose without a weight is valid
      return 
    ...
```

HDCharles committed:
Update on "Getting NS to work with Ferraris"


Summary: WIP

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32928449).

HDCharles committed:
Update on "Fixing bug with qconfig check"


Summary: The assert_valid_qconfig function tries to check whether a
qconfig on a convT model has a per_channel weight qscheme, but the NS
debug qconfig: torch.ao.quantization.QConfig(activation=logger_cls, weight=None)
results in an error when qconfig.weight() would be called. Added a convt
model to NS test to check for this in the future.

Test Plan: python test/test_quantization.py TestEagerModeNumericSuite.test_compare_model_stub_conv_static

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32928449](https://our.internmc.facebook.com/intern/diff/D32928449)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32928449).

HDCharles committed:
Update on "Fixing bug with qconfig check"


Summary: The assert_valid_qconfig function tries to check whether a
qconfig on a convT model has a per_channel weight qscheme, but the NS
debug qconfig: torch.ao.quantization.QConfig(activation=logger_cls, weight=None)
results in an error when qconfig.weight() would be called. Added a convt
model to NS test to check for this in the future.

Test Plan: python test/test_quantization.py TestEagerModeNumericSuite.test_compare_model_stub_conv_static

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32928449](https://our.internmc.facebook.com/intern/diff/D32928449)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32928449).
